src/framework/synchronization/index.ts

```diff
@@ -180,3 +189,33 @@ async function uploadData<T extends UploadDataLookup, K extends keyof T>(
     { concurrency: UPLOAD_CONCURRENCY },
   );
 }
+
+interface AbortSynchronizationInput extends SynchronizationJobContext {
+  reason?: string;
+}
+/**
+ * Aborts a synchronization job
+ */
+export async function abortSynchronization({
+  logger,
+  apiClient,
+  job,
+  reason,
+}: AbortSynchronizationInput) {
+  logger.info('Aborting synchronization job...');
+
+  let abortedJob: SynchronizationJob;
+
+  try {
+    const response = await apiClient.post(
+      `/persister/synchronization/jobs/${job.id}/abort`,
+      { reason },
+    );
+    abortedJob = response.data.job;
+  } catch (err) {
+    const errorMessage = 'Error occurred while aborting synchronization job.';
+    throw synchronizationApiError(err, errorMessage);
```

Comment:
I'm curious more than anything... why alloc a local var only to use it in one place?

Reply:
Oh, this is copy pasta'd from another function. I had a logger.error there, but with how errors are now handled, I ended up removing that. There's a little inconsistency in the sync functions, so lemme clean that up rnrq.